opencog/openpsi/utilities.scm

```diff
@@ -326,13 +326,13 @@
 		result
 		; else check if entity is an evaluation or predicate or schema
 		(let ((type (cog-type entity)))
-			(if (or (equal? type 'PredicateNode)
-				   (equal? type 'GroundedPredicateNode)
+			(if (or (equal? type 'GroundedPredicateNode)
 				   (equal? type 'DefinedPredicateNode))
 				(set! result (cog-evaluate! (Evaluation entity (List)))))
-			(if (or (equal? type 'SchemaNode)
-            		(equal? type 'GroundedSchemaNode)
-            	    (equal? type 'DefinedSchemaeNode))
+			(if (equal? type 'PredicateNode)
+				(set! result (cog-tv entity)))
```

Comment:
It seems to not matter which one to use...

PLN should have rules deriving one from the other ...

If not using PLN, it would be safest to just set both values...

On Mon, Aug 1, 2016 at 12:24 PM, Eddie Monroe <notifications@github.com>
wrote:

> In opencog/openpsi/utilities.scm
> <https://github.com/opencog/opencog/pull/2334#discussion_r73017565>:
>
> >  				   (equal? type 'DefinedPredicateNode))
> >  				(set! result (cog-evaluate! (Evaluation entity (List)))))
> > -			(if (or (equal? type 'SchemaNode)
> > -            		(equal? type 'GroundedSchemaNode)
> > -            	    (equal? type 'DefinedSchemaeNode))
> > +			(if (equal? type 'PredicateNode)
> > +				(set! result (cog-tv entity)))
>
> However, I think that for now it is better just to be careful to use these
> consistently
>
> Yes, that is my point!! And the question I was attempting to ask is which
> of these 2 possibilities should we be consistently using?!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencog/opencog/pull/2334/files/255ac40a46ca72d1fe5b4f88e3959a99a2c6658d#r73017565>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFolXMxI0rLiV4anr41qOlaKn-AIXOUyks5qbivbgaJpZM4JYxpE>
> .
>



-- 
Ben Goertzel, PhD
http://goertzel.org

Super-benevolent super-intelligence is the thought the Global Brain is
currently struggling to form...